packages/react-vapor/src/components/table-hoc/TableWithEmptyState.tsx

```diff
@@ -20,21 +22,32 @@ export const tableWithEmptyState = (
         return {
             isTrulyEmpty,
             data: isTrulyEmpty ? null : ownProps.data,
+            isEmptyStateAlreadySet: TableSelectors.isEmptyStateAlreadySet(state, ownProps),
         };
     };
 
+    const mapDispatchToProps = (dispatch: IDispatch, ownProps: ITableHOCOwnProps) => ({
+        setEmptyState: () => dispatch(TableHOCActions.setEmptyState(ownProps.id, true)),
+    });
+
     const TableEmptyState: React.FunctionComponent<
-        ITableHOCOwnProps & TableWithEmptyStateProps & ReturnType<typeof mapStateToProps>
+        ITableHOCOwnProps &
+            TableWithEmptyStateProps &
+            ReturnType<typeof mapStateToProps> &
+            ReturnType<typeof mapDispatchToProps>
     > = (props) => {
-        const {emptyState, isTrulyEmpty: isTrulyEmpty, ...tableProps} = props;
+        const {emptyState, isTrulyEmpty: isTrulyEmpty, setEmptyState, isEmptyStateAlreadySet, ...tableProps} = props;
         const [shouldRenderEmptyState, setShouldRenderEmptyState_immediate] = React.useState(false);
 
         const setShouldRenderEmptyState_debounced = React.useRef(
             _.debounce((value: boolean) => setShouldRenderEmptyState_immediate(value), UPDATE_DELAY)
         ).current;
 
         // Cancelling the debounced function on unmount to prevent calling setState on an unmounted component
-        React.useEffect(() => setShouldRenderEmptyState_debounced.cancel, []);
+        React.useEffect(() => {
+            !isEmptyStateAlreadySet && props.setEmptyState();
+            setShouldRenderEmptyState_debounced.cancel;
+        }, []);
```

Comment:
I am pretty sure that you are breaking something here. 
What a `useEffect` is returning is executed on unmount.

So the initial implementation was not executing anything on mount, but only on unMount, because the only thing that the `useEffect` did is to return the cancel function. But in your implementation, you do not return anything.

IMO you should create another useEffect with your dependency.

```suggestion
        React.useEffect(() => setShouldRenderEmptyState_debounced.cancel, []);
        React.useEffect(() => {
            if(!isEmptyStateAlreadySet) {
                props.setEmptyState();
            }
        },[isEmptyStateAlreadySet])
```